locallib.php

```diff
@@ -2245,6 +2245,19 @@ function bigbluebuttonbn_output_recording_table($bbbsession, $recordings, $tools
     return html_writer::div(html_writer::table($table), '', array('id' => 'bigbluebuttonbn_recordings_table'));
 }
 
+/**
+ * Helper function renders recording link for opencast.
+ *
+ * @param string $courseid
+ *
+ * @return array
+ */
+function bigbluebuttonbn_output_recording_opencast($courseid) {
+    $opencasturl = new \moodle_url('/blocks/opencast/index.php', array('courseid' => $courseid));
+    return html_writer::div(html_writer::tag('button', get_string('view_message_oc_recordings', 'bigbluebuttonbn'),
+                                array('class' => 'btn btn-primary', 'onclick' => "window.location='{$opencasturl}';")),
+                                 '', array('id' => 'bigbluebuttonbn_recordings_opencast', 'class' => 'py-3'));
```

Comment:
I think that these lines which are there to output the Opencast recordings in the BBB meeting activity have to be redone from scratch.

I see three major problems with this implementation:

1. These lines just render a buttons which links to the opencast overview page on /blocks/opencast/index.php. As you see on https://github.com/Opencast-Moodle/moodle-block_opencast/blob/master/index.php#L44, this page requires the `block/opencast:viewunpublishedvideos` capability which is assigned by default to editing teachers and managers only. Thus, students who click this link will just end up on an access denied page.
2. The button which you are rendering is opening the link with an JavaScript onclick handler. This is bad practice. You can better buttons with the `single_button` output renderer. Please see https://github.com/Opencast-Moodle/moodle-block_opencast/blob/master/index.php#L187-L188 for an example.
3. The most important problem is that, at least to my understanding, the user does not see a table of recordings which he could click and play. Instead, the user is redirected to /blocks/opencast/index.php but there is no guarantee that the user will see a list of _playable_ videos there. There has been https://github.com/Opencast-Moodle/moodle-block_opencast/pull/112/files which _could_ create links to playable videos on blocks/opencast/index.php, but this has to be configured as well.

What I would propose is that you:
1. Add a public function to block_opencast which lets you fetch the list of videos, their metadata and their play links for a given course.
2. Here, call this function and try to fetch the list of videos for this course.
3. If the returned list contains at least one video, filter the list for videos which were created by the BBB meeting activity (based on the BBB session ID).
4. If there are videos from this BBB meeting activity, render a recordings table similar to the table which is rendered for BBB native recordings.

Reply:
The new recording display feature is now available, which was pretty confusing and required lots of effort. please take a look and let me know your feedback, thanks you.